nemo/collections/asr/models/classification_models.py

```diff
@@ -138,9 +151,36 @@ def transcribe(self, paths2audio_files: List[str], batch_size: int = 4, logprobs
             logging.set_verbosity(logging_level)
         return labels
 
+    def prepare_manifest(self, config):
+        manifest_vad_input = config.get('manifest_vad_input', "manifest_vad_input.json")
+        input_audios = []
+        with open(config['manifest_filepath'], 'r') as manifest:
+            for line in manifest.readlines():
+                input_audios.append(json.loads(line.strip()))
+
+        p = Pool(processes=config['num_workers'])
+        args_func = {
+            'label': 'infer',
+            'split_duration': config['split_duration'],
+            'time_length': config['time_length'],
+        }
+        results = p.starmap(write_manifest_data, zip(input_audios, repeat(args_func)))
+        p.close()
+
+        if os.path.exists(manifest_vad_input):
+            logging.info("The prepared manifest file exists. Overwriting!")
+            os.remove(manifest_vad_input)
+
+        with open(manifest_vad_input, 'a') as fout:
```

Comment:
Is there a necessity to write under DDP? Cant we have a script that preprocesses this outside of model?